set image from geopackage

GeoAI Qgis plugin v. 3.0
I noticed that it is not possible to use an image saved inside a geopackage.

Failed to set image: OpenCV(4.11.0) D:\bld\libopencv_1743559191847\work\modules\imgproc\src\color.cpp:199: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'

If I export the image, the problem does not occur.
Thank you

--- a/qgis_plugin/geoai_plugin/dialogs/samgeo.py
+++ b/qgis_plugin/geoai_plugin/dialogs/samgeo.py
@@ -1042,8 +1042,19 @@ class SamGeoDockWidget(QDockWidget):
             if temp_export_path and os.path.exists(temp_export_path):
                 try:
                     os.remove(temp_export_path)
-                except Exception:
-                    pass
+                    # Keep internal temp file tracking consistent, if present
+                    if hasattr(self, "_temp_files"):
+                        try:
+                            self._temp_files.remove(temp_export_path)
+                        except ValueError:
+                            # It was not tracked; ignore
+                            pass
+                except Exception as cleanup_error:
+                    # Log cleanup failure but do not mask the original error
+                    self.log_message(
+                        f"Failed to clean up temporary export file '{temp_export_path}': {cleanup_error}",
+                        level=Qgis.Warning,
+                    )
             self.show_error(f"Failed to set image: {str(e)}")
 
         finally:

